grabbit/src/main/groovy/com/twcable/grabbit/server/batch/steps/jcrnodes/JcrNodesProcessor.groovy

```diff
@@ -88,10 +90,51 @@ class JcrNodesProcessor implements ItemProcessor<JcrNode, Node> {
             }
         }
 
-        final List<JcrProperty> properties = jcrNode.properties.toList()
-        Node.Builder nodeBuilder = Node.newBuilder()
+        // It checks if the node parent has required primary type as "nt:hierarchyNode"
+        // and then skip it as the node has already been processed by parent node
+        if(isRequiredHierarchyChildNode(jcrNode)) {
+            return null;
+        } else if(isHierarchyNode(jcrNode)) {
+            // Parent "nt:hierarchyNode" type will be processed
+            return buildNode(jcrNode, getRequiredChildNodes(jcrNode))
+        } else {
+            // Non-Hierarchy node logic
+            return buildNode(jcrNode, null)
+        }
+    }
+
+    private Node buildNode(JcrNode jcrNode, Collection<JcrNode> childNodes) {
+        final Node.Builder nodeBuilder = Node.newBuilder()
         nodeBuilder.setName(jcrNode.path)
+        nodeBuilder.setProperties(buildProperties(jcrNode))
+        childNodes?.each {
+            nodeBuilder.addChildNode(buildNode(it, getRequiredChildNodes(it)))
```

Comment:
I understand what it does.  What I'm asking is if this is useful if all known node types require *zero* or more (the zero ore more part being critical, as opposed to 1 or more) other mandatory nt:hierarchyNode types.  Part of Jim's algorithm is just to satisfy these nodes when we get to them.

This would be necessary if a node type were to exist such that it required 1 or more children of type nt:hierarchyNode